src/main/java/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition.java

```diff
@@ -198,25 +254,118 @@ public AutoCompletionCandidates doAutoCompleteJobName(@QueryParameter String val
-
-            ListBoxModel r = new ListBoxModel();
+            if (jobName != null) {
+                // JENKINS-25011: also look for jobs in folders.
+                j = getItemByPath(jobName, defaultJob, AbstractProject.class);
+            }
+           
             if (j!=null) {
                 JobPropertyImpl pp = j.getProperty(JobPropertyImpl.class);
                 if (pp!=null) {
-                    for (PromotionProcess proc : pp.getActiveItems())
+                    for (PromotionProcess proc : pp.getActiveItems()) {
+                        // Note: why not list all items instead of active ones?
+                        // this would allow to configure the job even
+                        // if a promotion hasn't happened (yet).
                         r.add(new Option(proc.getDisplayName(),proc.getName()));
+                    }
                 }
             }
             return r;
         }
     }
-
+    
+    //TODO: get relative paths based on folders?
+    //TODO: move to a helper class and then to the Jenkins core
+    @CheckForNull
+    @SuppressWarnings("unchecked")
+    private static <T extends Item> T getItemByPath(@Nonnull String path, @CheckForNull Item currentItem, 
+            @Nonnull Class<T> type) {
+        final Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins == null) {
+            return null;
+        }
+        
+        // Legacy behavior
+        TopLevelItem topLevelItem = jenkins.getItem(path);
+        if (topLevelItem != null && type.isAssignableFrom(topLevelItem.getClass())) {
+            return (T)topLevelItem;
+        }
+        
+        // Explicit global addressing
+        if (path.startsWith("/")) {
+            return findPath(jenkins, path.substring(1), type);
+        }
+        
+        // Try the relative addressing if possible
+        if (currentItem != null) {
+            final ItemGroup<?> relativeRoot = currentItem instanceof ItemGroup<?>
+                    ? (ItemGroup<?>)currentItem : currentItem.getParent();
+            
+            // Explicit relative addressing with "./", implicit 
+            final String effectivePath = path.startsWith("./") ? path.substring(2) : path;
+            final T item = findPath(relativeRoot, effectivePath, type);
+            if (item != null) {
+                return item;
+            }
+        }
+        
+        // Fallback to the default behavior
+        return findPath(jenkins, path, type);
+    }
+    
+    @CheckForNull
+    @SuppressWarnings("unchecked")
+    private static <T extends Item> T findPath(ItemGroup base, String path, Class<T> type) {
+        Item item = findPath(base, path);
+        if (item != null && type.isAssignableFrom(item.getClass())) {
+            return (T) item;
+        }
+        return null;
+    }
+    
+    @CheckForNull
+    private static Item findPath(ItemGroup base, String path) {
```

Comment:
> already exists and is designed and tested for the job

Could you send me a link to the implementation? I'll see what can I do on this direction